source/fuzz/fuzzer_pass_split_blocks.cpp

```diff
@@ -50,39 +51,51 @@ void FuzzerPassSplitBlocks::Apply() {
       continue;
     }
     // We are going to try to split this block.  We now need to choose where
-    // to split it.  We do this by finding a base instruction that has a
-    // result id, and an offset from that base instruction.  We would like
-    // offsets to be as small as possible and ideally 0 - we only need offsets
-    // because not all instructions can be identified by a result id (e.g.
-    // OpStore instructions cannot).
-    std::vector<std::pair<uint32_t, uint32_t>> base_offset_pairs;
+    // to split it.  We describe the instruction before which we would like to
+    // split a block via the opcode 'opc' of the relevant instruction, a base
+    // instruction 'base' that has a result id, and the number of instructions
+    // with opcode 'opc' that we should skip when searching from 'base' for the
+    // desired instruction.  (When the instruction before which we would like to
+    // split the block actually has a result id, the instruction is used as
+    // base, 'opc' is its opcode, and there are 0 instructions to skip.)
+    std::vector<std::tuple<uint32_t, SpvOp, uint32_t>> base_opcode_skip_triples;
+
     // The initial base instruction is the block label.
     uint32_t base = block->id();
-    uint32_t offset = 0;
+
+    // Counts the number of times we have seen each opcode since we reset the
+    // base instruction.
+    std::map<SpvOp, uint32_t> skip_count;
```

Comment:
As discussed offline, I would first find the instruction you want and then use a utility function that works backwards to find the nearest result id. This way, we don't need to interleave the logic for finding a suitable instruction and the logic that keeps track of the nearest result id. Plus, the utility function can be trivially re-used.

Reply:
I have opened https://github.com/KhronosGroup/SPIRV-Tools/issues/2964 to track this - I agree we should do this a simpler way (and quite possibly the way you propose), but indeed in a future PR.